tests/ducktape/test_consumer.py

```diff
@@ -147,73 +185,245 @@ def test_basic_consume(self, consumer_type, batch_size):
+                rebalance_events.append(len(partitions))
+                await consumer.assign(partitions)
 
-        self.logger.info(f"Testing {consumer_type} consumer polling (single messages) for {test_duration} seconds")
-
-        # Start metrics collection
-        metrics.start()
-
-        # Container for consumed messages
-        consumed_messages = []
-
-        # Run the test using poll_messages instead of consume_messages
-        start_time = time.time()
-        messages_consumed = strategy.poll_messages(
-            topic_name, test_duration, start_time, consumed_messages, timeout=0.1
-        )
-
-        # Finalize metrics collection
-        metrics.finalize()
-
-        # Get comprehensive metrics summary
-        metrics_summary = metrics.get_summary()
-        is_valid, violations = validate_consumer_metrics(metrics_summary, bounds)
-
-        # Print comprehensive metrics report
-        print_consumer_metrics_report(metrics_summary, is_valid, violations, consumer_type, 1)
-
-        # Enhanced assertions using metrics
-        assert messages_consumed > 0, "No messages were consumed"
-        assert len(consumed_messages) > 0, "No messages were collected"
-        assert metrics_summary['messages_consumed'] > 0, "No messages were consumed (metrics)"
-        assert metrics_summary['consumption_rate_msg_per_sec'] > 0, \
-            f"Consumption rate too low: {metrics_summary['consumption_rate_msg_per_sec']:.2f} msg/s"
-
-        # Validate against performance bounds
-        if not is_valid:
-            self.logger.warning("Performance bounds validation failed: %s", "; ".join(violations))
+            try:
+                # Phase 1: Consumer1 joins (should get all partitions)
+                await consumer1.subscribe([topic_name], on_assign=track_rebalance)
+                await self._wait_for_assignment(consumer1, expected_partitions=2)
+                assert len(rebalance_events) == 1
 
-        self.logger.info("Successfully completed basic poll test with comprehensive metrics")
+                # Phase 2: Consumer2 joins (should split partitions)
+                await consumer2.subscribe([topic_name], on_assign=track_rebalance)
+                await self._wait_for_balanced_assignment([consumer1, consumer2], total_partitions=2)
+                assert len(rebalance_events) >= 2
+
+                # Phase 3: Consumer2 leaves (consumer1 should get all partitions back)
+                await consumer2.close()
+                await self._wait_for_assignment(consumer1, expected_partitions=2)
+                assert len(rebalance_events) >= 3
+
+                # Verify functionality
+                self.produce_test_messages(topic_name, num_messages=1)
+                msg = await consumer1.poll(timeout=5.0)
+                assert msg is not None, "Consumer should receive fresh message"
+
+            finally:
+                await consumer1.close()
+
+        asyncio.run(async_rebalance_test())
```

Comment:
A quick look at ducktape looked like it doesn't support async test defs to avoid this wrapper. Maybe we could make an async supporting wrapper class that does this in the Metaclass? Could contribute it back to enable async tests in ducktape later on. Can be a follow-up task -- don't block this PR on it